### DISCUSSION TITLE
a very little mistake in the code example-maze.R

### DISCUSSION THREAD

**Discussion by bonushenricus** [andrewmarx/samc#19] [Q&A]
Hello. The small error, not of code as such, but of reference, is at line 193, which I imagine that instead of `plot_maze (maze_occ," Occupancy ", vir_col)` it should be `plot_maze (maze_occ_3," Occupancy ", vir_col)`
The question is so stupid that it's actually more of an excuse to introduce myself: I'm an agricultural technician, and in a research project I'm monitoring the spread in the agro-ecosystem of a parasitoid hymenoptera we launched a year ago.
SAMC's update work and the examples I find are absolutely fantastic and relevant to the work I am trying to do. But is it mandatory to enter the destination?

**Comment by andrewmarx** [comment]
Thanks, I'll fix that when I get a chance.

For the `dest` parameter, whether or not it is required depends on the metric. `cond_passage()`, for example, absolutely requires it. For something like `visitation()`, it's completely optional, and excluding it will just return results for all possible destinations, which may or may not be what you want. The long-term variant of `dispersal()` is a special case where not having `dest` will result in very long computation times, so it's better to use it if you don't need the results for all possible destinations.

All the possible input combinations should be in the documentation for the metrics. So depending on what metrics you're interested in, the documentation will show you any options that don't require `dest`
